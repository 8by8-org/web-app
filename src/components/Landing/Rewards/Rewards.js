import CurveA from "./../../../assets/2-shapes/curve-a.svg";
import './Rewards.scss';

const Rewards = () => {
    // Consistent data type
    // Also includes key since we will be iterating over it
    class Partner {
        static key = 0;
        constructor(name, img, type, location) {
            this.name = name;
            this.img = img;
            this.type = type;
            this.location = location;
            this.id = Partner.key++;
        }
    }

    
    // Sample Data
    const samplePartner = () => new Partner (
        "Joe's Sandwhiches",
        "https://c8.alamy.com/comp/FC20MT/sandwich-shop-in-milan-FC20MT.jpg",
        "Restaurant",
        "San Jose, CA"
    )

    // Store the list of partners here
    // Generate them with the constructor
    const partners = [
        samplePartner(),
        samplePartner(),
        samplePartner(),
        samplePartner()
    ]


    return (
        <div className="Rewards">
            <div className="bg-black pt-32px pr-30px pl-30px pb-30px">
                <h1>
                    Rewards
                </h1>
                <p className="description-text">
                    Win the 8by8 challenge, and you'll recieve a reward from one of our amazing partners!
                </p>
            </div>
            <img className="curve" src={CurveA} alt="black curve" />
            <div className="sponsor-section">
                {partners.map((partner) => {
                    return (
                        <div className="sponsor" key={partner.id}>
                            <img src={partner.img} alt={`${partner.name} image`} />
                            <h5>{partner.name}</h5>
                            <p>{partner.type} {partner.location}</p>
                        </div>
                    )
                })}
            </div>
        </div>
    )
}

export default Rewards;