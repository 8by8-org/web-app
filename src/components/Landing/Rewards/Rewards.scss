@import "../../../styles/Components.scss";
@import "../../../styles/Fonts.scss";
@import "../../../styles/Variables.scss";
@import "../../../styles/Classes.scss";

.Rewards {
    h1 {
        @include text-underline;
        color: white;
        font-size: 4rem;
        text-transform: capitalize;
    }

    .description-text {
        color: white;
    }

    .sponsor-section {
        margin: 20px 0px;
        display: flex;
        flex-direction: column;
        align-items: center;
        
        img {
            margin: 10px 0px;
            height: 20vh;
            width: 30vh;
        }

        .sponsor {
            text-align: center;
            margin: 20px 0px;
            display: flex;
            flex-direction: column;
            align-items: center;
            a {
                text-decoration: none !important;
            }
            h5 {
                
                font-size: 2rem;
                font-family: $bebas-neue;
                text-transform: uppercase;
            }
            p {
                margin: 0 2rem;
            }
            
            .challenge-button {
                margin: 2rem;
                @include button;
                width: 80% !important;
            }
        }
    }
}