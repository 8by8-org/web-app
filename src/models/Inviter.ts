export type Inviter = {
  name:string;
  avatar:'1' | '2' | '3' | '4';
  inviterId:string;
}