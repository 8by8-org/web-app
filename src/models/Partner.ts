import {StaticImageData} from "next/image";

export class Partner {
    static key = 0;

    partnerName:string;
    partnerLink:string;
    partnerImg:StaticImageData | null | string;
    partnerType:string;
    partnerLoc:string;
    partnerID:number;

    constructor(name:string, link:string, img:StaticImageData | null, type:string, location:string) {
        this.partnerName = name;
        this.partnerLink = link;
        this.partnerImg = img;
        this.partnerType = type;
        this.partnerLoc = location;
        this.partnerID = Partner.key++;
    }
}