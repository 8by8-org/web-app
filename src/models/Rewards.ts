export type Rewards = Array<{
  businessDescription:string;
  businessLink:string;
  businessType:string;
  locationDescription:"Online" | "In Person";
  locationType:"Online" | "In Person";
  logo:string;
  name:string;
  redemptionDescription:string;
  rewardAvailable:boolean;
  rewardConditions:string;
  rewardDescription:string;
  rewardEndDate:Date | undefined;
  rewardLink:string;
  rewardStartDate:Date;
  rewardType:"Online" | "In Person";
}>
