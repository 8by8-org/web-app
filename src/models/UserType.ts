export enum UserType {
  PLAYER = "player",
  CHALLENGER = "challenger",
  HYBRID = "hybrid"
}