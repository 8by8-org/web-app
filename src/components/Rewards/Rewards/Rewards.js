import CurveA from "../../../assets/2-shapes/curve-a.svg";
import './Rewards.scss';
import { useHistory } from "react-router-dom";
import placeholderImage from "../../../assets/images/placeholder-image.jpg"

const Rewards = () => {
    const history = useHistory();

    // Consistent data type
    // Also includes key since we will be iterating over it
    class Partner {
        static key = 0;
        constructor(name, link, img, type, location) {
            this.name = name;
            this.link = link;
            this.img = img;
            this.type = type;
            this.location = location;
            this.id = Partner.key++;
        }
    }


    // Sample Data
    const samplePartner = () => new Partner(
        "Joe's Sandwhiches",
        "#",
        "",
        "Restaurant",
        "San Jose, CA"
    )

    // Store the list of partners here
    // Generate them with the constructor
    // That way there is a key
    const partners = [
        samplePartner(),
        samplePartner(),
        samplePartner(),
        samplePartner()
    ]


    return (
        <div className="Rewards">
            <div className="bg-black pt-32px pr-30px pl-30px pb-30px">
                <h1>
                    Rewards
                </h1>
                <p className="description-text">
                    Win the 8by8 challenge, and you'll recieve a reward from one of our amazing partners!
                </p>
            </div>
            <img className="curve" src={CurveA} alt="black curve" />
            <div className="sponsor-section">
                { /* If we have partners */}
                {partners.length > 0 && partners.map((partner) => {
                    return (
                        <div className="sponsor" key={partner.id}>
                            <a href={partner.link}>
                                <img src={partner.img === "" ? placeholderImage : partner.img} alt={`${partner.name} image`} />
                                <h5>{partner.name}</h5>
                                <p>{partner.type} {partner.location}</p>
                            </a>
                        </div>
                    )
                })}

                { /* If we don't have any partners */}
                {partners.length === 0 && (
                    <div className="sponsor">
                        <img src={placeholderImage} alt="Placeholder image" />
                        <p>
                            Unfortunately, we are not offering partner rewards at this time. You can still take the 8by8 Challenge to help the AAPI community!
                        </p>
                        <button
                            className="challenge-button"
                            onClick={() => history.push("/challengerwelcome")}
                        >
                            Take the Challenge!
                        </button>
                    </div>
                )}
            </div>
        </div>
    )
}

export default Rewards;