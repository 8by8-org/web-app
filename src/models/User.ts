import { UserType } from "./UserType";

export type User = {
  name:string;
  avatar:'1' | '2' | '3' | '4';
  type:UserType;
  registeredVoter:boolean;
  notifyElectionReminders:boolean;
  startedChallenge:boolean;
  playerReward:string;
  playerState:string;
}