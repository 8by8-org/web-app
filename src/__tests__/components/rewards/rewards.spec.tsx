import { useRouter } from 'next/router';
import mockRouter from 'next-router-mock';

import { render, screen, cleanup, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'
import Rewards from "@/pages/rewards";
import {Partner} from "@/models/Partner";
import chefusLogo from "../../../assets/partner-logos/chefus.png";
import purelyLogo from "../../../assets/partner-logos/purely.png";
import saltalkLogo from "../../../assets/partner-logos/saltalk.png";
import tippsyLogo from "../../../assets/partner-logos/tippsy.png";

// Sample Data
const Chefus = new Partner(
    "Chefus",
    "https://www.chefus.com/",
    chefusLogo,
    "Online deliveries",
    "",
   )

const Purely = new Partner(
    "Purely Drinks",
    "https://purelydrinks.com/",
    purelyLogo,
    "Online deliveries",
    "",
)

const Saltalk = new Partner(
    "Saltalk",
    "https://www.saltalk.com/",
    saltalkLogo,
    "Online deliveries",
    "",
)

const Tippsy = new Partner(
    "Tippsy",
    "https://www.tippsysake.com/",
    tippsyLogo,
    "Online deliveries",
    "",
)

const FakePartner = new Partner(
    "Fake",
    "https://www.example.com/",
    null,
    "Online deliveries",
    "",
)


// Store the list of partners here
// Generate them with the constructor
// That way there is a key
const partnerArray = [
   Chefus, Purely, Saltalk, Tippsy
]

jest.mock('next/router', () => require('next-router-mock'));

describe('Rewards', () => {
  afterEach(() => cleanup());

  it('renders rewards component', () => {
    const {container} = render(<Rewards partners={[]}></Rewards>);

    expect(container).toMatchSnapshot();
  });

  it('mocks the useRouter hook', () => {
    // Set the initial url:
    //mockRouter.push("/initial-path");
    
    // Render the component:
    render(<Rewards partners={[]} />);

    // Click the button:
    fireEvent.click(screen.getByRole('button'));
    
    // Ensure the router was updated:
    expect(mockRouter.asPath).toEqual(('/challengerwelcome'));
  });

  it('renders rewards component with partners', () => {
    const {container} = render(<Rewards partners={[Chefus, Purely, Saltalk, Tippsy, FakePartner]}></Rewards>);

    expect(container).toMatchSnapshot();
  });
});