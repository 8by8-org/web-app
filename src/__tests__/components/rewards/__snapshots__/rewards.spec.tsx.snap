// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rewards renders rewards component 1`] = `
<div>
  <div
    class="Rewards"
  >
    <div
      class="bg-black pt-32px pr-30px pl-30px pb-30px"
    >
      <h1>
        Rewards
      </h1>
      <p
        class="description-text"
      >
        Win the 8by8 challenge, and you'll recieve a reward from one of our amazing partners!
      </p>
    </div>
    <img
      alt="black curve"
      class="curve"
      src="[object Object]"
    />
    <div
      class="sponsor-section"
    >
      <div
        class="sponsor"
      >
        <img
          alt="Placeholder"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <p>
          Unfortunately, we are not offering partner rewards at this time. You can still take the 8by8 Challenge to help the AAPI community!
        </p>
        <button
          class="challenge-button"
        >
          Take the Challenge!
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Rewards renders rewards component with partners 1`] = `
<div>
  <div
    class="Rewards"
  >
    <div
      class="bg-black pt-32px pr-30px pl-30px pb-30px"
    >
      <h1>
        Rewards
      </h1>
      <p
        class="description-text"
      >
        Win the 8by8 challenge, and you'll recieve a reward from one of our amazing partners!
      </p>
    </div>
    <img
      alt="black curve"
      class="curve"
      src="[object Object]"
    />
    <div
      class="sponsor-section"
    >
      <div
        class="sponsor"
      >
        <a
          href="https://www.chefus.com/"
        >
          <img
            alt="Chefus"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <h5>
            Chefus
          </h5>
          <p>
            Online deliveries
             
          </p>
        </a>
      </div>
      <div
        class="sponsor"
      >
        <a
          href="https://purelydrinks.com/"
        >
          <img
            alt="Purely Drinks"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <h5>
            Purely Drinks
          </h5>
          <p>
            Online deliveries
             
          </p>
        </a>
      </div>
      <div
        class="sponsor"
      >
        <a
          href="https://www.saltalk.com/"
        >
          <img
            alt="Saltalk"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <h5>
            Saltalk
          </h5>
          <p>
            Online deliveries
             
          </p>
        </a>
      </div>
      <div
        class="sponsor"
      >
        <a
          href="https://www.tippsysake.com/"
        >
          <img
            alt="Tippsy"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <h5>
            Tippsy
          </h5>
          <p>
            Online deliveries
             
          </p>
        </a>
      </div>
      <div
        class="sponsor"
      >
        <a
          href="https://www.example.com/"
        >
          <img
            alt="Fake"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <h5>
            Fake
          </h5>
          <p>
            Online deliveries
             
          </p>
        </a>
      </div>
    </div>
  </div>
</div>
`;
